File: board-room/src/main/java/net/stzups/board/data/objects/canvas/object/CanvasObjectWrapper.java
package net.stzups.board.data.objects.canvas.object;

import io.netty.buffer.ByteBuf;

public class CanvasObjectWrapper {
    private byte dt;
    private CanvasObject canvasObject;

    public CanvasObjectWrapper(CanvasObjectType canvasObjectType, ByteBuf byteBuf) {
        dt = byteBuf.readByte();
        canvasObject = CanvasObject.getCanvasObject(canvasObjectType, byteBuf);
    }

    public CanvasObject getCanvasObject() {
        return canvasObject;
    }

    public void serialize(ByteBuf byteBuf) {
        byteBuf.writeByte(dt);
        canvasObject.serialize(byteBuf);
    }
}

File: board-frontend/src/scripts/canvas/CanvasObjectWrapper.js
import {getCanvasObject} from "./Canvas.js";

export default class CanvasObjectWrapper {
    constructor(type, reader) {
        this.dt = reader.readUint8();
        this.canvasObject = getCanvasObject(type, reader);
    }

    serialize(writer) {
        writer.writeUint8(this.dt);
        this.canvasObject.serialize(writer);
    }

    static create(dt, canvasObject) {
        let object = Object.create(this.prototype);
        object.dt = dt;
        object.canvasObject = canvasObject;
        return object;
    }
}